packages/fx-core/src/core/index.ts

```diff
@@ -277,6 +281,40 @@ export class FxCore implements Core {
     return ok(projectPath);
   }
 
+  async archive(projectPath: string): Promise<void> {
+    const archiveFolderPath = path.join(projectPath, ArchiveFolderName);
+    await fs.ensureDir(archiveFolderPath);
+
+    const fileNames = await fs.readdir(projectPath);
+    const archiveLog = async (projectPath: string, message: string): Promise<void> => {
+      await fs.appendFile(
+        path.join(projectPath, ArchiveLogFileName),
+        `[${new Date().toISOString()}] ${message}\n`
+      );
+    };
+
+    await archiveLog(projectPath, `Start to move files into '${ArchiveFolderName}' folder.`);
+    for (const fileName of fileNames) {
+      if (fileName === ArchiveFolderName || fileName === ArchiveLogFileName) {
+        continue;
+      }
+
+      try {
+        await fs.move(path.join(projectPath, fileName), path.join(archiveFolderPath, fileName), {
+          overwrite: true,
+        });
+      } catch (e: any) {
+        await archiveLog(projectPath, `Failed to move '${fileName}'. ${e.message}`);
+        throw e;
```

Comment:
Just a consideration, should we make archiving an atomic operation?

Reply:
The first design is to copy all the files into `.archive` folder, then delete the files in the root folder. In this way the operation can be atomic (using rollback).
But copy operation takes a lot of time(node_modules more than 5 minutes), and move operation is fast (less than 1 second).
So we use a log file to let the user known what happened.

Reply:
Make sense.